homeassistant/components/ambee/sensor.py

```diff
@@ -62,26 +46,18 @@ def __init__(
         *,
         coordinator: DataUpdateCoordinator,
         entry_id: str,
-        sensor_key: str,
-        sensor: AmbeeSensor,
+        description: SensorEntityDescription,
         service_key: str,
         service: str,
     ) -> None:
         """Initialize Ambee sensor."""
         super().__init__(coordinator=coordinator)
-        self._sensor_key = sensor_key
+        self._sensor_key = description.key
         self._service_key = service_key
 
-        self.entity_id = f"{SENSOR_DOMAIN}.{service_key}_{sensor_key}"
-        self._attr_device_class = sensor.get(ATTR_DEVICE_CLASS)
-        self._attr_entity_registry_enabled_default = sensor.get(
-            ATTR_ENABLED_BY_DEFAULT, True
-        )
-        self._attr_icon = sensor.get(ATTR_ICON)
-        self._attr_name = sensor.get(ATTR_NAME)
-        self._attr_state_class = sensor.get(ATTR_STATE_CLASS)
-        self._attr_unique_id = f"{entry_id}_{service_key}_{sensor_key}"
-        self._attr_unit_of_measurement = sensor.get(ATTR_UNIT_OF_MEASUREMENT)
+        self.entity_id = f"{SENSOR_DOMAIN}.{service_key}_{description.key}"
+        self._attr_entity_description = description
```

Comment:
I'm wondering if it would be better to have a method call here instead that assigns the attributes based on the description contents. Especially since some of them (e.g., `self._attr_name`) might need to be overwritten.
```py
        self._set_attributes_from_description(description)
        ...
```

Reply:
They can be overwritten with the current solution. Additionally, buy keeping it, allows for integrations to extend the description with extra information they need/want.

This prevent the needs to overwrite the method on those occurrences and just access it.